.github/workflows/main.yml

```diff
@@ -18,7 +18,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout the code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3

```

Comment:
**question**: Why do we change to v3 and what does it change?

(same question for `upload-artifact`)

Reply:
Checkout and upload-artifact 3.0 update node js to [16.0](https://github.com/nodejs/node/releases/tag/v16.0.0).
This version of node js including performance tweaks and improvements.